File: inst/expt/rob.R
library(cvxr)

set.seed(1234)
x <- matrix(rnorm(100), nrow=25)

S <- cov(x)

delta <- 0.1

alpha <- Variable(rows = 4)
e1 <- c(1, 0 , 0, 0)
e2 <- c(0, 1 , 0, 0)
e3 <- c(0, 0 , 1, 0)
e4 <- c(0, 0 , 0, 1)

objective <- Minimize(0.5 * quad_form(x = alpha, P = S)  + t(alpha) %*% e2 + delta * Pnorm(alpha, 1))
p <- Problem(objective)
solution <- solve(p)

File: inst/expt/rob.py
#! /usr/bin/env python
from cvxpy import *
import canonInterface
import numpy

x  = numpy.random.randn(4, 10)
A = numpy.cov(x)
#S = Semidef(4)
S = A

delta = 0.1
alpha = Variable(4)

e1 = numpy.array([1, 0, 0, 0])
e2 = numpy.array([0, 1, 0, 0])
e3 = numpy.array([0, 0, 1, 0])
e4 = numpy.array([0, 0, 0, 1])

S = numpy.genfromtxt("s.csv", delimiter=",")

objective = Minimize(0.5 * quad_form(alpha, S) + alpha.T * e2 + delta * norm(alpha, 1))
#constraints = [ alpha.T * S * alpha >= 0]
prob = Problem(objective)
prob.solve()
